skimage/measure/_regionprops_utils.py

```diff
@@ -326,3 +328,34 @@ def perimeter_crofton(image, directions=4):
 
     total_perimeter = coefs @ h
     return total_perimeter
+
+def _normalize_spacing(spacing:Any, ndims:int):
+    """Normalizes the spacing parameter
+
+    The `spacing` parameter should be a tuple of numbers
+    that is the same length as the number of dimensions
+    in the image.
+    If the spacing is a single number (e.g. float)
+    then assume spacing is same in all dimensions.
+
+    Parameters
+    ---------
+    spacing : Any
+        User-provided `spacing` keyword.
+    ndims: int
+        The number of dimensions in the image
+
+    Returns
+    -------
+    spacing : tuple of floats
+        Corrected spacing, if possible.
+    """
+
+    if (np.ndim(spacing) == 1) and (len(spacing) == ndims):
+        if not np.all([np.isreal(x) for x in spacing]):
+            raise ValueError(f'Spacing should be a tuple of {ndims} floats.')
+        return np.array(spacing)
+    elif isinstance(spacing, Real):
+        return np.full(ndims, float(spacing))
+    else:
+        raise ValueError(f'Spacing should be a tuple of {ndims} floats.')
```

Comment:
This is still using `np.real` in case `sequence` has the shape `(ndims,)`. Maybe we can simplify this somewhat. How about

```suggestion
    spacing = np.array(spacing)
    if spacing.shape == ():
        spacing = np.broadcast_to(spacing, shape=(ndims,))
    elif spacing.shape != (ndims,):
        raise ValueError(
            f"spacing isn't a scalar or a sequence of shape {(ndims,)}, got {spacing}"
        )
    if not all(isinstance(s, Real) for s in spacing):
        raise TypeError(
            f"spacing isn't of floating or integer type, got {spacing}"
        )
    return spacing
```

This will also require updating the tests below, because a `TypeError` is raised instead of a `ValueError`.